src/Services/Air/Air.js

```diff
@@ -284,21 +284,15 @@ module.exports = (settings) => {
 
     cancelPNR(options) {
       return this.getUniversalRecordByPNR(options)
-        .then(
-          ur => getBookingFromUr(ur, options.pnr)
-            || Promise.reject(new AirRuntimeError.NoPNRFoundInUR(ur))
-        )
-        .catch(
-          err => Promise.reject(new AirRuntimeError.FailedToCancelPnr(options, err))
-        )
-        .then((record) => {
-          return (
-            record.tickets.length === 0
-              ? Promise.resolve([])
-              : this.getTickets({ reservationLocatorCode: record.uapi_reservation_locator })
-          )
-            .then(
-              tickets => Promise.all(tickets.map(
+        .then((ur) => {
+          const record = Array.isArray(ur) ? ur[0] : ur;
+
+          return this.getTickets(record)
+            .catch(() => {
```

Comment:
If the ticket couldn't be retrieved you will get an error message like this.

```xml
<SOAP:Envelope 
  xmlns:SOAP="http://schemas.xmlsoap.org/soap/envelope/">
  <SOAP:Body>
    <air:AirRetrieveDocumentRsp TransactionId="F54E52500A07405C5494698BE234E2CA" ResponseTime="90" 
      xmlns:air="http://www.travelport.com/schema/air_v39_0" 
      xmlns:common_v39_0="http://www.travelport.com/schema/common_v39_0">
      <common_v39_0:ResponseMessage Code="12009" Type="Error" ProviderCode="1G">Host error during ticket retrieve.- Ticket Number - UNABLE TO PROCESS INVALID TICKET NUMBER</common_v39_0:ResponseMessage>
    </air:AirRetrieveDocumentRsp>
  </SOAP:Body>
</SOAP:Envelope>
```
This will generate error for you.
Sometimes you can get error because of uAPI outage.

If we just return empty array here we could cancel segments with still active ticket.

Either remove this check with returning empty array, or add additional param `ignoreTickets=false|true` to the `cancelPNR` method to skip tickets check. This way you can skip the check in case of error.